types/got/index.d.ts

```diff
@@ -98,6 +87,37 @@ declare namespace got {
         (url: GotUrl, options: GotBodyOptions<null>): GotPromise<Buffer>;
     }
 
+    interface GotJSONFn {
+        (url: GotUrl): GotPromise<any>;
+        (url: GotUrl, options: GotJSONOptions): GotPromise<any>;
+    }
+
+    type GotInstance<T = GotFn> = T &
+        Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', T> &
+        {
+            stream: GotStreamFn & Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', GotStreamFn>;
+            extend: GotExtend;
+            RequestError: typeof RequestError;
+            ReadError: typeof ReadError;
+            ParseError: typeof ParseError;
+            HTTPError: typeof HTTPError;
+            MaxRedirectsError: typeof MaxRedirectsError;
+            UnsupportedProtocolError: typeof UnsupportedProtocolError;
+            CancelError: typeof CancelError;
+            TimeoutError: typeof TimeoutError;
+        };
+
+    type DefaultGotInstance = GotInstance;
```

Comment:
I think the following type aliases are redundant for public use cases:

* `GotInstance` can be used instead `DefaultGotInstance`.
* `GotInstance<ConcreteType>` can be used instead `JSONGotInstance`.

Reply:
👍